dot/rpc/subscription/websocket.go

```diff
@@ -159,25 +164,24 @@ func (c *WSConn) initStorageChangeListener(reqID float64, params interface{}) (L
 		wsconn: c,
 	}
 
-	pA, ok := params.([]interface{})
-	if !ok {
-		return nil, fmt.Errorf("unknown parameter type")
-	}
-	for _, param := range pA {
-		switch p := param.(type) {
-		case []interface{}:
-			for _, pp := range param.([]interface{}) {
-				data, ok := pp.(string)
-				if !ok {
-					return nil, fmt.Errorf("unknown parameter type")
-				}
-				stgobs.filter[data] = []byte{}
+	switch filter := params.(type) {
+	case string:
+		stgobs.filter[filter] = []byte{}
```

Comment:
why do we set this to an empty slice and not `nil`? Setting it to `nil` would still set the key in the map, and we can then set it or append to it as well later.

Reply:
Just a few bytes less. I guess not a big deal either, so up to you!